core/light-client/src/state.rs

```diff
@@ -44,14 +45,21 @@ impl<Block: BlockT> RelayState<Block> {
 	pub fn new(block_header: Block::Header, validator_set: AuthorityList) -> Self {
 		RelayState {
 			header_hashes: vec![block_header.hash()],
-			last_finalized_block_header: block_header,
+			last_finalized_block_header: block_header.clone(),
+			// is it bad to initialize with the same? Header trait does no implement default...
+			penultimate_finalized_block_header: block_header,
```

Comment:
Strictly speaking the `penultimate_finalized_block_header` is an `Option`, no? Because when it is not guaranteed to have a meaningful value.

Reply:
Yes, I have been thinking about that too. But there is literally only one case when we don't have a penultimate header, and this is block number 0. So, I don't think it's worth introducing an extra case here.

Reply:
That's why it's called an 'edge case' 😉 do we have a test for that?
I know it feels tedious, but from my experience, it's exactly these small short cuts and surprises that come back to haunt you and a developer spends hours debugging.

Reply:
Hmm, I will leave it currently as is for now. This piece of code will only be relevant once sidechain block import is activated. Depending on how we will do that, it is not even necessary. There is a Todo tagged with #423 in the light-client for that.